pkg/pgmodel/ingestor/dispatcher.go

```diff
@@ -132,52 +167,83 @@ func (p *pgxDispatcher) runCompleteMetricCreationWorker() {
 }
 
 func (p *pgxDispatcher) runSeriesEpochSync() {
-	epoch, err := p.refreshSeriesEpoch(model.InvalidSeriesEpoch)
-	// we don't have any great place to report errors, and if the
-	// connection recovers we can still make progress, so we'll just log it
-	// and continue execution
-	if err != nil {
-		log.Error("msg", "error refreshing the series cache", "err", err)
-	}
+	p.refreshSeriesEpoch()
 	for {
 		select {
 		case <-p.seriesEpochRefresh.C:
-			epoch, err = p.refreshSeriesEpoch(epoch)
-			if err != nil {
-				log.Error("msg", "error refreshing the series cache", "err", err)
-			}
+			p.refreshSeriesEpoch()
 		case <-p.doneChannel:
 			return
 		}
 	}
 }
 
-func (p *pgxDispatcher) refreshSeriesEpoch(existingEpoch model.SeriesEpoch) (model.SeriesEpoch, error) {
-	dbEpoch, err := p.getServerEpoch()
+func (p *pgxDispatcher) getDatabaseEpochs() (model.SeriesEpoch, model.SeriesEpoch, error) {
+	var dbCurrentEpoch, dbDeleteEpoch int64
+	row := p.conn.QueryRow(context.Background(), getEpochSQL)
+	err := row.Scan(&dbCurrentEpoch, &dbDeleteEpoch)
+	if err != nil {
+		return 0, 0, err
+	}
+	return model.SeriesEpoch(dbCurrentEpoch), model.SeriesEpoch(dbDeleteEpoch), nil
+}
+
+func (p *pgxDispatcher) refreshSeriesEpoch() {
+	log.Info("msg", "Refreshing series cache epoch")
+	cacheCurrentEpoch := p.scache.CacheEpoch()
+	dbCurrentEpoch, dbDeleteEpoch, err := p.getDatabaseEpochs()
+	log.Info("msg", "refreshSeriesEpoch", "cacheCurrentEpoch", cacheCurrentEpoch, "dbCurrentEpoch", dbCurrentEpoch, "dbDeleteEpoch", dbDeleteEpoch)
 	if err != nil {
+		log.Warn("msg", "Unable to get database epoch data, will reset series and inverted labels caches", "err", err.Error())
 		// Trash the cache just in case an epoch change occurred, seems safer
-		p.scache.Reset()
+		p.scache.Reset(dbCurrentEpoch)
 		// Also trash the inverted labels cache, which can also be invalidated when the series cache is
 		p.invertedLabelsCache.Reset()
-		return model.InvalidSeriesEpoch, err
+		return
+	}
+	if cacheCurrentEpoch.After(dbCurrentEpoch) {
+		log.Warn("msg", "The connector's cache epoch is greater than the databases. This is unexpected", "connector_current_epoch", cacheCurrentEpoch, "database_current_epoch", dbCurrentEpoch)
```

Comment:
If we do flush the cache, it would probably make sense to also reset the cache's epoch to the epoch we just got from the DB. Seems like a sensible move.